### PR TITLE
append error message to end of help

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

To save scrolling back past the full help text to see what the problem was, print any error messages at the end (after the help). #1651

### CODE REVIEW

**Comment by shemnon** [comment]
Tangent: why do we even show the help for random errors?  Shouldn't they be limited to option parsing errors?

File: artemis/src/main/java/tech/pegasys/artemis/cli/BeaconNodeCommand.java [hunk 1]
```diff
@@ -217,12 +219,33 @@ private int handleParseException(final CommandLine.ParameterException ex, final
...
+      appendErrorMessageToHelp(ex.getCommandLine(), ex.getMessage());
       ex.getCommandLine().usage(outputWriter, CommandLine.Help.Ansi.AUTO);
     }
 
     return ex.getCommandLine().getCommandSpec().exitCodeOnInvalidInput();
   }
 
+  private void appendErrorMessageToHelp(CommandLine cmd, String message) {
+    final String SECTION_KEY_NEW_HEADING = "errorHeading";
+    final String SECTION_KEY_NEW_DETAILS = "errorContent";
```
Comment (ajsutton) [inline artemis/src/main/java/tech/pegasys/artemis/cli/BeaconNodeCommand.java]: nit: These are not constants so should use normal camel case (or be moved to actual constants).

**Review by ajsutton — APPROVED** [review]
This is definitely an improvement but my understanding was that we were moving to not show the help for most errors (and just make the error text helpful).

File: artemis/src/main/java/tech/pegasys/artemis/cli/BeaconNodeCommand.java (hunk 1 above)
Comment (macfarla) [reply artemis/src/main/java/tech/pegasys/artemis/cli/BeaconNodeCommand.java]: fixed

**Review by MadelineMurray — COMMENTED** [review]
Text LGTM

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**